server/node/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/view/widget/RecordPanel.java

```diff
@@ -235,6 +242,28 @@ public void onFailure(Throwable caught) {
         }
         
     }
+
+    private class CustomFormDataLoader implements FormDataLoader {
+
+        @Override
+        public void loadFormData(String fileItemName,
+                                 final AsyncCallback<RecordField> callback) {
+            String schema = recordFieldWidget.getValue().getSchema();
+            KaaAdmin.getDataSource().getDataFromFile(schema, fileItemName,
+                    new AsyncCallback<RecordField>() {
+                        @Override
+                        public void onSuccess(RecordField result) {
+                            callback.onSuccess(result);
+                        }
+
+                        @Override
+                        public void onFailure(Throwable caught) {
+                            callback.onFailure(caught);
+                        }
+                    });
+        }
+
+    }
     
```

Comment:
Rename class CustomFormDataLoader to ConfigurationFormDataLoader and move it to ConfigFormStructView